File: lin.c
#include <stdio.h>
#include <time.h>
int main(){
    //X=(A*X+C)%M
    long x=0,a=214013,c=2531011,m=0x80000000-1,j=0;
    clock_t t=clock();
    x=c;
    while(x!=0){x=(a*x+c)%m;j++;}
    printf("%lf[s]:%ld[1]\n",(double)(clock()-t)/CLOCKS_PER_SEC,j);
    return 0;
}

File: lin.go
package main

import "time"

func main() {
	//X=(A*X+C)%M
	var x = 0
	var a = 214013
	var c = 2531011
	var m = 0x80000000 - 1
	var j = 0
	var t = time.Now().UnixNano()
	x = c
	for x != 0 {
		x = (a*x + c) % m
		j++
	}
	print((time.Now().UnixNano()-t)/int64(time.Microsecond), "[μs]", j, "[1]")
}
